### PR TITLE
added solution hw-24.2

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by aleksandr-hrankin — APPROVED** [review]

File: src/main/java/com/internet/shop/dao/OrderDao.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public interface OrderDao {
+
+    Order create(Order order);
+
+    Optional<Order> get(Long id);
+
+    boolean delete(Long id);
+
+    List<Order> getAllByUserId(Long userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/OrderDao.java]:
```suggestion
    List<Order> getUserOrders(Long userId);
```

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ShoppingCart dbShoppingCart = shoppingCartDao.get(shoppingCart.getId()).get();
+        dbShoppingCart.getProducts().remove(product);
+        return shoppingCartDao.update(dbShoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ShoppingCart dbShoppingCart = shoppingCartDao.get(shoppingCart.getId()).get();
+        dbShoppingCart.getProducts().clear();
+        shoppingCartDao.update(dbShoppingCart);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        shoppingCart.getProducts().clear();
        shoppingCartDao.update(dbShoppingCart);
```
it's better not to call db twice, we can expect that cart exists

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        ShoppingCart dbShoppingCart = shoppingCartDao.get(shoppingCart.getId()).get();
+        dbShoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(dbShoppingCart);
+    }
+
+    @Override
+    public ShoppingCart deleteProduct(ShoppingCart shoppingCart, Product product) {
+        ShoppingCart dbShoppingCart = shoppingCartDao.get(shoppingCart.getId()).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: same here

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        ShoppingCart dbShoppingCart = shoppingCartDao.get(shoppingCart.getId()).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: and here

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        dbShoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(dbShoppingCart);
+    }
+
+    @Override
+    public ShoppingCart deleteProduct(ShoppingCart shoppingCart, Product product) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
```
i think it's better if we keep boolean as return type and return list.remove() result

**Review by RomanDubovskyi — COMMENTED** [review]
good job, left minor comments

File: src/main/java/com/internet/shop/dao/OrderDao.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/dao/OrderDao.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/OrderService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.internet.shop.service;
+
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+
+@Service
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/OrderService.java]: this annotation is not needed here

**Review by RomanDubovskyi — APPROVED** [review]
remove redundant annotation

File: src/main/java/com/internet/shop/service/OrderService.java (hunk 6 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/service/OrderService.java]: done

**Pull request merged**